cluster-autoscaler/simulator/predicates_checker_interface.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package simulator
+
+import (
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	scheduler_nodeinfo "k8s.io/kubernetes/pkg/scheduler/nodeinfo"
+)
+
+// PredicateChecker checks whether all required predicates pass for given Pod and Node.
+type PredicateChecker interface {
+	// TODO remove
+	SnapshotClusterState() error
+	// TODO drop "nodeInfos map[string]*scheduler_nodeinfo.NodeInfo" and check against schedulerLister
```

Comment:
Not sure about this TODO -> I probably want FitsAnyNode(snapshot, pod, candidateNodes), not just FitsAnyNode(snapshot, pod). I expect this to come up a lot in scale-down, where you have an explicit pod destination nodes list and you may also not want to reschedule to another candidate node, etc.

Reply:
It could be useful interface. Yet currently the method is only used by filter-out-schedulable which checks pod agains all nodes in the cluster. 
Will leave it for now for further extension if we change scale-down to make use of it.

Reply:
I'm not going to let you ignore this one without any explanation (though we should probably continue discussion in my comment on scheduler_based_predicates_checker.go, which makes the same point in more detail).